HW_2/task_4.py

```diff
@@ -0,0 +1,21 @@
+"""
+Задайте список из N элементов, заполненный целыми числами из промежутка [-N, N].
+Найдите произведение элементов на индексах, хранящихся в файле indexes.txt (в одной строке один индекс).
+Решение должно работать при любом натуральном N.
+
+Ввод: значение типа <int>
+Вывод: значение типа <int>
+"""
+
+from random import randint
+
+number = int(input("Введите натуральное число: "))
+my_list = [randint(-number, number)for _ in range(number)]
+print(my_list)
+res = 1
+with open('indexes.txt', 'r') as file:
+    for line in file:
+        index = int(line)
+        if len(my_list) > index >= -len(my_list):
```

Comment:
У вас же уже есть переменная, хранящая длину списка.
```suggestion
        if number > index >= -number:
```